converter/input/mgologstash/flow_deserializer.go

```diff
@@ -62,6 +72,119 @@ func (f *flowDeserializer) updateExporterUptimesMap(ipfixMap bson.M, host string
+		startMillsIface, startMillsOk := ipfixMap["flowStartSysUpTime"]
+		if !startMillsOk {
+			return false
+		}
+		//try converting to an int64 first, handle any errors that come up
+		startMills, startMillsOk = startMillsIface.(int64)
+		if !startMillsOk {
+			startMills32, start32Ok := startMillsIface.(int)
+			if !start32Ok {
+				return false
+			}
+			startMills = int64(startMills32)
+		}
+
+		//if the host's first flow milliseconds is greater than the new flow's
+		//  start milliseconds it implies that the system was reinitialized and it
+		//  is imparitive to update the information currently saved
+		if f.ipfixExporterRelUp[host].firstFlowMills > startMills {
+			newExporter, err := getNewExporterUptime(ipfixMap)
+			if err != nil {
+				return false
+			}
+
+			f.ipfixExporterRelUp[host] = newExporter
+			return true
+		}
+	} else {
+		newExporter, err := getNewExporterUptime(ipfixMap)
+		if err != nil {
+			return false
+		}
+		f.ipfixExporterRelUp[host] = newExporter
+
+		return true
+	}
+
+	return false
+}
+
+//Since the code for updating the relative system uptime and creating a new
+//relative system uptime are similar make a function so we don't repeat code
+//It returns a structure defined as ipfixRelTime (relative uptime) and any error
+// this code experiences
+func getNewExporterUptime(ipfixMap bson.M) (ipfixRelTime, error) {
+	emptyRelTime := ipfixRelTime{0.0, time.Now(), false}
+
+	//set the inital value if it hasn't been set
+	var startMills int64
+	startMillsIface, startMillsOk := ipfixMap["flowStartSysUpTime"]
+	//if the milliseconds since the init isn't present return false
+	if !startMillsOk {
+		return emptyRelTime, errors.Errorf("Couldn't find flowStartSysUpTime")
+	}
+	//try converting to an int64 first, handle any errors that come up
+	startMills, startMillsOk = startMillsIface.(int64)
+	if !startMillsOk {
+		startMills32, start32Ok := startMillsIface.(int)
+		if !start32Ok {
+			return emptyRelTime, errors.Errorf("Couldn't convert values")
+		}
+		startMills = int64(startMills32)
+	}
+
+	//get the timestamp value, if we can, then convert it to a time value
+	var flowDate time.Time
+	flowDateIface, flowDateOk := ipfixMap["timestamp"]
+	if flowDateOk {
+		//convert the time to a string, so we can parse the string to a date/time vlaue
+		flowDateStr, ok := flowDateIface.(string)
+		if !ok {
+			return emptyRelTime, errors.Errorf("Couldn't convert timestamp to string")
+		}
+		var err error
+		flowDate, err = time.Parse(time.RFC3339, flowDateStr)
+		if err != nil {
+			return emptyRelTime, err
+		}
+	} else {
+		//if we can't find the timestamp value assume the flow came now and use
+		//  that value
+		flowDate = time.Now()
+	}
+
+	return ipfixRelTime{startMills, flowDate, true}, nil
+}
+
+//getFlowUptimeMillis64 will take a flow uptime (milliseconds) as an interface
+//  and do the conversion to get it as an int64 for calculations
+func getFlowUptimeMillis64(flowUptimeMillisIface interface{}) (int64, error) {
```

Comment:
This function could be renamed castToInt64.

We might want to restructure this function as
getInt64FromMap(ipfixMap bson.M, key string) (int64, error).

This would remove the existence check from the call sites as well. 

Either way this function could be used throughout this chunk of code.

Going one step further, I think we should invest time working on a wrapping type for bson.M that does these typed accesses for us.
Ie. we would write a new struct UntypedMap. 
```
myMap := UntypedMap(ipfixMap)
flowStart, err := myMap.getInt64("flowStartSysUpTime")
//myMap also has getString, getInt16, getTime etc
```